/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.12/threading.py", line 1073, in _bootstrap_inner
    self.run()
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 1192, in run
    pubsub.get_message(ignore_subscribe_messages=True, timeout=sleep_time)
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 1054, in get_message
    response = self.parse_response(block=(timeout is None), timeout=timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 865, in parse_response
    response = self._execute(conn, try_read)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 841, in _execute
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/retry.py", line 65, in call_with_retry
    fail(error)
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 843, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 830, in _disconnect_raise_connect
    raise error
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/retry.py", line 62, in call_with_retry
    return do()
           ^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 842, in <lambda>
    lambda: command(*args, **kwargs),
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/client.py", line 859, in try_read
    if not conn.can_read(timeout=timeout):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/connection.py", line 570, in can_read
    return self._parser.can_read(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 79, in can_read
    return self.read_from_socket(timeout=timeout, raise_on_timeout=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fardin/Frappe/practice-frappe/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 90, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
